Update for eForms

duncandewhurst opened the pull request.

duncandewhurst committed:
Update for eForms

odscjen commented on README.md:
```diff
@@ -21,7 +21,7 @@ Here are the possible values for a framework agreement's `method` field, and com
 ## Legal context
 
-In the European Union, this extension's fields correspond to [eForms BG-706 (Techniques)](https://docs.ted.europa.eu/eforms/latest/reference/business-terms/). See [OCDS for the European Union](http://standard.open-contracting.org/profiles/eu/master/en/) for the correspondences to Tenders Electronic Daily (TED).
+In the European Union, this extension's fields correspond to [eForms BG-706 (Techniques)](https://docs.ted.europa.eu/eforms/latest/reference/business-terms/). For correspondences to eForms fields, see [OCDS for eForms](https://standard.open-contracting.org/profiles/eforms/latest/). For correspondences to Tenders Electronic Daily (TED), see [OCDS for the European Union](http://standard.open-contra ...
```
```suggestion
In the European Union, this extension's fields correspond to [eForms BG-706 (Techniques), BG-557 (Group Framework Maximum Value and BT-271 (Framework Maximum Value)](https://docs.ted.europa.eu/eforms/latest/reference/business-terms/). For correspondences to eForms fields, see [OCDS for eForms](https://standard.open-contracting.org/profiles/eforms/latest/). For correspondences to Tenders Electronic Daily (TED), see [OCDS for the European Union](http://standard.open-contracting.org/profiles/eu/master/en/).
```

odscjen commented on release-schema.json:
```diff
@@ -125,6 +134,11 @@
         },
+        "maximumValue": {
+          "title": "Value",
+          "description": "The estimated maximum value of the framework agreement, as estimated when publishing the tender information.",
```
Should the description of `value` also have a secondary sentence part describing that it should be used at the award and contract stages?

duncandewhurst replied:
Hmm, techniques only appear in the `tender` section, does it make sense to put actual award/contract values there?

jpmckinney replied:
OCDS 1.2 has `Award.maximumValue` for framework agreements (also, `Award.estimatedValue`).

odscjen replied:
tbh I'm now not sure what I was getting at with this comment! I think the point was that "The total upper estimated value of the framework agreement" and "The estimated maximum value of the framework agreement," are identical in meaning, it's only the second part of the description of `maximumValue` that differentiates it from `value`. So the definition of `value` needs something to make it clearer why both of these fields exist and which to use in any specific circumstance?

jpmckinney replied:
Ah, actually do we need to add `maximumValue` then? Is `value` used for another mapping? We can update the `value` description.

odscjen replied:
So there are 2 fields relevant here -  
* BT-271 (Framework Maximum Value) which we've mapped to `techniques.frameworkAgreement.maximumValue` 
* BT-157 (Group Framework Maximum Value) which we've mapped to `techniques.frameworkAgreement.value`.

BT-157 is only in `guidance.yaml` for LotsGroup, but BT-271 has both a Lot and a LotsGroup version.  

Looking at the definitions:

BT-157 = "The maximum value which may be spent in a framework agreement within a group of lots. This information can be provided when the maximum value of a group of lots is lower than the sum of maximum values of individual lots in this group (e.g. when the same budget is shared for several lots).  Maximum value means a value covering all contracts to be awarded within a framework agreement over its whole duration, including options and renewals."

BT-271 = "The maximum value of the framework agreement for the procurement procedure or lot, over its whole duration, including options and renewals. This value covers all contracts to be awarded within the framework agreement." The [guidance](https://docs.ted.europa.eu/eforms/latest/schema/all-in-one.html#estimatedValueSection) also states that this field "represents the early evaluation of the Framework Maximum value for the Group of Lots, the Lot or the Procedure as a whole."

The difference (in my reading) is that BT-271 is the aggregated value of all the lots in the group, where BT-157 may be less than the aggregated value if they are sharing a budget? So it is possible that over the course of related notices both of these values could be used which is why we added in the `maximumValue` field? In which case we do need both but I think the definition of at least one needs updating.

jpmckinney replied:
I think BT-271 matches the existing `value` field. It's the maximum value of the FA for the procedure (tender) or lot.

BT-157 is only for lot groups. So that would map to LotGroup.maximumValue (not sure if it's already in use).

odscjen replied:
It is, BT-27-LotsGroup (Estimated Value) is currently mapped to `lotGroups.maximumValue`

jpmckinney replied:
Do you have an idea what the difference is, in semantics, between BT-27-LotsGroup and BT-157-LotsGroup?

At any rate, BT-157-LotsGroup seems more like `maximumValue` and `BT-27-LotsGroup` more like `value`.

jpmckinney replied:
Ah, BT-157 is specific to framework agreements, but BT-27 is not.

jpmckinney replied:
This analysis got a bit long so I reopened https://github.com/open-contracting/european-union-support/issues/121

duncandewhurst replied:
I've removed `FrameworkAgreement.maximumValue` per https://github.com/open-contracting/european-union-support/issues/121#issuecomment-1588935142

duncandewhurst committed:
Update README.md

Co-authored-by: odscjen <95221058+odscjen@users.noreply.github.com>

jpmckinney committed:
Merge branch 'master' into eforms

odscjen commented on README.md:
```diff
@@ -21,7 +21,7 @@ Here are the possible values for a framework agreement's `method` field, and com
 ## Legal context
 
-In the European Union, this extension's fields correspond to [eForms BG-706 (Techniques)](https://docs.ted.europa.eu/eforms/latest/reference/business-terms/). See [OCDS for the European Union](http://standard.open-contracting.org/profiles/eu/master/en/) for the correspondences to Tenders Electronic Daily (TED).
+In the European Union, this extension's fields correspond to [eForms BG-706 (Techniques), BG-557 (Group Framework Maximum Value and BT-271 (Framework Maximum Value)](https://docs.ted.europa.eu/eforms/latest/reference/business-terms/). For correspondences to eForms fields, see [OCDS for eForms](https://standard.open-contracting.org/profiles/eforms/latest/). For correspondences to Tenders Electroni ...
```
```suggestion
In the European Union, this extension's fields correspond to [eForms BG-706 (Techniques), BG-157 (Group Framework Maximum Value and BT-271 (Framework Maximum Value)](https://docs.ted.europa.eu/eforms/latest/reference/business-terms/). For correspondences to eForms fields, see [OCDS for eForms](https://standard.open-contracting.org/profiles/eforms/latest/). For correspondences to Tenders Electronic Daily (TED), see [OCDS for the European Union](http://standard.open-contracting.org/profiles/eu/master/en/).
```
Sorry I think I introduced a typo

duncandewhurst committed:
release-schema.json: Remove FrameworkAgreement.maximumValue

Per https://github.com/open-contracting/european-union-support/issues/121#issuecomment-1588935142

duncandewhurst committed:
Update README.md

Co-authored-by: odscjen <95221058+odscjen@users.noreply.github.com>

jpmckinney committed:
Merge master into eforms
